src/api/comments.ts

```diff
@@ -0,0 +1,37 @@
+export const BASE_URL = 'https://mate.academy/students-api/';
+
+export async function getComments(postId: number) {
+  const response = await fetch(`${BASE_URL}/comments?postId=${postId}`);
+
+  if (!response.ok) {
+    throw new Error(`${response.status} - ${response.statusText}`);
+  }
+
+  return response.json();
```

Comment:
personally i prefer positive checks over negative checks if it is possible. I think it is possible to miss negative check while reading the code. So for me it is a matter of taste and readablility:
```suggestion
  if (response.ok) {
    return response.json();
  }
  
  throw new Error(`${response.status} - ${response.statusText}`);
```

Reply:
good point in checking `response.ok`. `fetch` promise does not reject on HTTP errors. This means that 4xx and 5xx response status code won't throw an error and you need to handle it by yourself. You should do it in all methods where you preform fetch. 

as an option you can have some general method, wich will be a wrapper around fetch for your project. It can get arguments the same way as fetch does but can throw an error or handle errors in way you want for all request.

Reply:
please check try/catch and response.ok check for all methods that do reqeusts